c10/metal/special_math.h

```diff
@@ -451,5 +451,15 @@ inline float digamma(T0 x) {
   }
 }
 
+template <typename T>
+T sinc(T a) {
+  if (a == static_cast<T>(0)) {
+    return static_cast<T>(1);
+  }
+  constexpr T pi = static_cast<T>(3.14159265358979323846L);
```

Comment:
Why not use M_PI_F here?

Reply:
I didn't remember about it. Let me change :)